src/creature_states_spdig.c

```diff
@@ -1285,17 +1285,17 @@ short imp_drops_gold(struct Thing *spdigtng)
         if (is_thing_directly_controlled_by_player(spdigtng, my_player_number))
         {
             play_non_3d_sample(119);
+            internal_set_thing_state(spdigtng, state);
```

Comment:
Why would we set a state to a creature when it is possessed?

Reply:
To keep it under possession, if I recall correctly.

Reply:
This does not sound correct to me. Possession overwrites states, the states are used to direct creature behavior.
I think it was there (but outside of this if) to get imps to get back to mining when they fail to drop off gold anywhere.

Reply:
I may need to test again, but if I remember correctly, if the Imp is under possession, he'll do his own thing even though he's under possession, unless his state is reset. We do not want possessed Imps to be set to the last work state.

Reply:
But you are setting possessed imps to last_work_state here right?

Reply:
No. Possessed Imps are set to the null state.